### PR TITLE
Lxh

### PR DESCRIPTION
**Pull request by innerlee**

### CODE REVIEW

File: examples/dcgan.py [hunk 1]
```diff
@@ -117,9 +117,9 @@
 n_col     = opt.nCol
 valbs     = n_row * n_col
 z_draw_np = latent.sample(valbs).float()
-z_draw    = torch.as_tensor(z_draw_np).cuda(async=True)
+z_draw    = z_draw_np.cuda(non_blocking=True)
```
Comment (innerlee) [inline examples/dcgan.py:120]: `blah.cuda()` has been replaced by `blah.to(device)` in new version of pytorch? see https://pytorch.org/docs/stable/notes/cuda.html#device-agnostic-code. maybe we can change to that style. what do u think?

File: examples/wgan-gp.py [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+
+#region Args for Training
+parser.add_argument('--gpu',            type=int, default=0, help='which GPU to use, default to 0')
+parser.add_argument('--nIter',          type=int, default=50000, help='number of iteration to train for')
+parser.add_argument('--repeatD',        type=int, default=5, help='repeat D per iteration')
+parser.add_argument('--repeatG',        type=int, default=1, help='repeat G per iteration')
+parser.add_argument('--optimizerG',     default='adam', help='adam | rmsprop | sgd, optimizer for G')
+parser.add_argument('--optimizerD',     default='adam', help='adam | rmsprop | sgd, optimizer for D')
+parser.add_argument('--lrG',            type=float, default=0.0001, help='learning rate for Generator, default=0.0001')
+parser.add_argument('--lrD',            type=float, default=0.0001, help='learning rate for Distance, default=0.0001')
```
Comment (innerlee) [inline examples/wgan-gp.py]: Distance -> Discriminator

File: examples/wgan-gp.py [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+
+#endregion yapf: enable
+
+#region Preparation
+
+os.environ['CUDA_VISIBLE_DEVICES'] = str(opt.gpu)
+cudnn.benchmark = True
+
+if opt.workdir is None:
+    opt.workdir = f'samples/wgan-gp/exp73_{datetime.now()}'.replace(' ', '_')
```
Comment (innerlee) [inline examples/wgan-gp.py]: exp73 -> exp

File: examples/dcgan.py [hunk 4]
```diff
@@ -90,7 +90,7 @@
 cudnn.benchmark = True
 
 if opt.workdir is None:
-    opt.workdir = f'samples/exp73_{datetime.now()}'.replace(' ', '_')
+    opt.workdir = f'samples/dcgan/exp73_{datetime.now()}'.replace(' ', '_')
```
Comment (innerlee) [inline examples/dcgan.py]: exp73 -> exp

File: examples/wgan-gp.py [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+        z_np = latent.sample(opt.bs).float()
+        with torch.no_grad():
+            z = z_np.cuda(non_blocking=True)
+            x_fake = netG(z)
+        x_fake = Variable(x_fake.data)
+
+        loss_fake = torch.mean(netD(x_real))
+        loss_real = -torch.mean(netD(x_fake))
+
+        batch_size = x_real.size()[0]
```
Comment (innerlee) [inline examples/wgan-gp.py]: is this same as `x_real.size(0)`?

File: examples/wgan-gp.py [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+        gradients = grad(outputs=pred_hat, inputs=x_hat, grad_outputs=torch.ones(pred_hat.size()).cuda(),
+                         create_graph=True, retain_graph=True, only_inputs=True)[0]
+
+        loss_gp = opt.lambdaGP * ((gradients.view(gradients.size()[0], -1).norm(2, 1) - 1) ** 2).mean()
+
+        loss_fake.backward()
+        loss_real.backward()
+        loss_gp.backward(retain_graph=True)
+
+        prob_D_real += np.exp(-loss_real.data.item())
```
Comment (innerlee) [inline examples/wgan-gp.py]: loss of wgan do not have a probabilistic meaning. u can output the losses directly

**Comment by innerlee** [comment]
the non-blocking modification can be done later
